### PR TITLE
feat(multichain): Add Polygon and Base

### PR DESCRIPTION
**Pull request by finnian0826**
### Description

Add Polygon and Base to wallet 

### Test plan

builds

everything behind feature gate

### Related issues

- Fixes #ACT-1121

### Backwards compatibility

Yes

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [X] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/nfts/NftsInfoCarousel.tsx [hunk 1]
```diff
@@ -151,6 +151,10 @@ export default function NftsInfoCarousel({ route }: Props) {
     [NetworkId['arbitrum-sepolia']]: t('viewOnArbiscan'),
     [NetworkId['op-mainnet']]: t('viewOnOPMainnetExplorer'),
     [NetworkId['op-sepolia']]: t('viewOnOPSepoliaExplorer'),
+    [NetworkId['polygon-mainnet']]: t('viewOnPolygonScan'),
+    [NetworkId['polygon-mumbai']]: t('viewOnPolygonScan'),
```
Comment (silasbw) [inline src/nfts/NftsInfoCarousel.tsx:155]: can you call this `polygon-pos-mumbai` and `polygon-pos-mainnet`?  That's what we've used elsewhere and it helps differentiate between the different polygon networks (there's a zkEVM-based network from polygon too).

File: src/transactions/types.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ export enum Network {
   Ethereum = 'ethereum',
   Arbitrum = 'arbitrum',
   Optimism = 'optimism',
+  Polygon = 'polygon',
```
Comment (silasbw) [inline src/transactions/types.ts:10]: `PolygonPoS` and `polygon-pos` here?

File: src/config.ts [hunk 3]
```diff
@@ -122,6 +122,16 @@ export const ALCHEMY_OPTIMISM_API_KEY = keyOrUndefined(
   DEFAULT_TESTNET,
   'ALCHEMY_OPTIMISM_API_KEY'
 )
+export const ALCHEMY_POLYGON_API_KEY = keyOrUndefined(
+  secretsFile,
+  DEFAULT_TESTNET,
+  'ALCHEMY_POLYGON_API_KEY'
```
Comment (silasbw) [inline src/config.ts:128]: the naming nit i mention below.  Maybe it's `ALCHEMY_POLYGON_POS_API_KEY` here?

**Review by silasbw — APPROVED** [review]

File: locales/base/translation.json [hunk 4]
```diff
@@ -2092,6 +2092,8 @@
   "viewOnArbiscan": "View on Arbiscan",
   "viewOnOPMainnetExplorer": "View on OP Mainnet Explorer",
   "viewOnOPSepoliaExplorer": "View on OP Sepolia Explorer",
+  "viewOnPolygonPoSScan": "View on PolygonPoSScan",
```
Comment (satish-ravi) [inline locales/base/translation.json:2095]: could value just be "View on PolygonScan"? `PolygonPoSScan` doesn't seem very readable for the user

File: src/shared/conts.ts [hunk 5]
```diff
@@ -9,4 +9,8 @@ export const NETWORK_NAMES: Record<NetworkId, string> = {
   [NetworkId['arbitrum-sepolia']]: 'Arbitrum Sepolia',
   [NetworkId['op-mainnet']]: 'Optimism',
   [NetworkId['op-sepolia']]: 'Optimism Sepolia',
+  [NetworkId['polygon-pos-mainnet']]: 'PolygonPoS',
+  [NetworkId['polygon-pos-mumbai']]: 'PolygonPoS Mumbai',
```
Comment (satish-ravi) [inline src/shared/conts.ts:13]: I believe this is another user facing string, is the PoS suffix essential cc @silasbw

**Review by satish-ravi — APPROVED** [review]

**Pull request merged**